server/applicationServer.go

```diff
@@ -35,9 +37,41 @@ func NewServer() *ApplicationServer {
 // Start will start the server eventually
 func (s *ApplicationServer) Start(addr string) error {
 	s.Config.ServerAddress = addr
-	s.httpServer.Addr = s.Config.ServerAddress
 
-	err := s.httpServer.ListenAndServe()
+	listner, err := net.Listen("tcp", s.Config.ServerAddress)
+	if err != nil {
+		log.Fatal("Listen:", err)
+	} else {
+		address := listner.Addr().String()
+		ip := ""
+		port := ""
+
+		arr := strings.Split(address, ":")
+		if strings.Contains(address, "[::]") {
+			ip = "127.0.0.1"
+			port = arr[len(arr)-1]
+		} else {
+			// IPV4
+			const numberOfItems int = 2
+			if len(arr) == numberOfItems {
+				ip = arr[0]
+				port = arr[1]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '1', create a named constant for it](https://app.codacy.com/gh/hrple/common/pullRequest?prid=5576416)